### DISCUSSION TITLE
Console Issue In my Dev Tools for Almost Amazon site

### DISCUSSION THREAD

**Discussion by fimoefive** [nss-evening-cohort-14/evening-client-side#131] [Help Me!] [answered]
_I'm getting a issues tab on my console in Dev Tools
The console issue is asking me to indicate where to send a cookie request by specifying an updated attributes of the cookie.
My google sign in table is timing out every time I click the red Google tab. I've reset my internet and closed out every application on my computer and restarted it twice before relaunching the Almost Amazon app.
Please help me with this console issue, this is way over my head and don't want to mess up my API or fetch calls._

Indicate whether to send a cookie in a cross-site request by specifying its SameSite attribute

Because a cookie’s SameSite attribute was not set or is invalid, it defaults to SameSite=Lax, which prevents the cookie from being sent in a cross-site request. This behavior protects user data from accidentally leaking to third parties and cross-site request forgery.
Resolve this issue by updating the attributes of the cookie:
Specify SameSite=None and Secure if the cookie should be sent in cross-site requests. This enables third-party use.
Specify SameSite=Strict or SameSite=Lax if the cookie should not be sent in cross-site requests.

Name | Domain & Path
-- | --
HSID | .google.com/
SSID | .google.com/
APISID | .google.com/
SAPISID | .google.com/
SID | .google.com/
SIDCC | .google.com/

Name | Domain & Path
-- | --
HSID | .google.com/
SSID | .google.com/
APISID | .google.com/
SAPISID | .google.com/
SID | .google.com/
SIDCC | .google.com/
api.js?onload=__iframefcb534070

**Comment by drteresavasquez** [comment]
Make sure you're not in incognito mode and also make sure you're running your server through local host when running npm start. 

If you've followed ALL the steps in both the readme and the videos please tell us the steps you did and at what point did you get errors.

**Reply by fimoefive** [reply]
I first went into Chrome settings and enabled to allow all Cookies and data in the Chrome browser.
I just updated my home-brew and zsh files and went to https://github.com/nss-nightclass-projects/Night-Class-Resources/blob/master/book-1-foundations/chapters/mac-installs.md to make sure I had everything installed. I also reEntered the npm install command in my terminal. Now when I do a "npm start" it fails to compile, now I'm in worse shape than I was yesterday evening.

    ERROR in ./src/styles/main.scss (./node_modules/css-loader/dist/cjs.js??ref--7-1!./node_modules/sass-loader/dist/cjs.js??ref--7-2!./src/styles/main.scss)
    Module build failed (from ./node_modules/sass-loader/dist/cjs.js):
    Error: Node Sass does not yet support your current environment: OS X 64-bit with Unsupported runtime (88)
    For more information on which environments are supported please see:
    https://github.com/sass/node-sass/releases/tag/v4.14.1
        at module.exports (/Users/mjsisk/workspace3/foundations/inclass/IN-CLASS-almost-amazon-template/node_modules/node-sass/lib/binding.js:13:13)
        at Object.<anonymous> (/Users/mjsisk/workspace3/foundations/inclass/IN-CLASS-almost-amazon-template/node_modules/node-sass/lib/index.js:14:35)
        at Module._compile (node:internal/modules/cjs/loader:1091:14)
        at Object.Module._extensions..js (node:internal/modules/cjs/loader:1120:10)
        at Module.load (node:internal/modules/cjs/loader:971:32)
        at Function.Module._load (node:internal/modules/cjs/loader:812:14)
        at Module.require (node:internal/modules/cjs/loader:995:19)
        at require (node:internal/modules/cjs/helpers:92:18)
        at getDefaultSassImplementation (/Users/mjsisk/workspace3/foundations/inclass/IN-CLASS-almost-amazon-template/node_modules/sass-loader/dist/utils.js:43:10)
        at getSassImplementation (/Users/mjsisk/workspace3/foundations/inclass/IN-CLASS-almost-amazon-template/node_modules/sass-loader/dist/utils.js:57:30)
ℹ ｢wdm｣: Failed to compile.

**Reply by drteresavasquez** [reply]
I think you are doing too much. The issue wasn't a local issue. It was a chrome issue. I'm not sure what to say other than work through the steps again and make sure you're reading your terminal. If you tried npm start with errors. What are they? Did you npm i?

**Reply by fimoefive** [reply]
I've ran npm i every time before entering npm start. 
This is what's showing up in my terminal when trying to run Almost Amazon

 IN-CLASS-almost-amazon-template git:(main) ✗ npm i

up to date, audited 1196 packages in 2s

found 0 vulnerabilities
➜  IN-CLASS-almost-amazon-template git:(main) ✗ npm start

> webpack-intro@1.0.0 start
> webpack-dev-server --mode development --open

ℹ ｢wds｣: Project is running at http://localhost:8081/
ℹ ｢wds｣: webpack output is served from /
ℹ ｢wds｣: Content not from webpack is served from /Users/mjsisk/workspace3/foundations/inclass/IN-CLASS-almost-amazon-template
ℹ ｢wdm｣: wait until bundle finished: /
✖ ｢wdm｣: Hash: bd6300dd9da2e5360a2b
Version: webpack 4.44.1
Time: 2385ms
Built at: 02/23/2021 3:20:51 PM
       Asset       Size  Chunks             Chunk Names
./index.html  281 bytes          [emitted]  
   bundle.js   3.11 MiB    main  [emitted]  main
Entrypoint main = bundle.js
[0] multi (webpack)-dev-server/client?http://localhost:8081 ./src/javascripts/main.js 40 bytes {main} [built]
[./node_modules/ansi-html/index.js] 4.16 KiB {main} [built]
[./node_modules/bootstrap/dist/js/bootstrap.js] 132 KiB {main} [built]
[./node_modules/webpack-dev-server/client/index.js?http://localhost:8081] (webpack)-dev-server/client?http://localhost:8081 4.29 KiB {main} [built]
[./node_modules/webpack-dev-server/client/overlay.js] (webpack)-dev-server/client/overlay.js 3.51 KiB {main} [built]
[./node_modules/webpack-dev-server/client/socket.js] (webpack)-dev-server/client/socket.js 1.53 KiB {main} [built]
[./node_modules/webpack-dev-server/client/utils/createSocketUrl.js] (webpack)-dev-server/client/utils/createSocketUrl.js 2.91 KiB {main} [built]
[./node_modules/webpack-dev-server/client/utils/log.js] (webpack)-dev-server/client/utils/log.js 964 bytes {main} [built]
[./node_modules/webpack-dev-server/client/utils/reloadApp.js] (webpack)-dev-server/client/utils/reloadApp.js 1.59 KiB {main} [built]
[./node_modules/webpack-dev-server/client/utils/sendMessage.js] (webpack)-dev-server/client/utils/sendMessage.js 402 bytes {main} [built]
[./node_modules/webpack-dev-server/node_modules/strip-ansi/index.js] (webpack)-dev-server/node_modules/strip-ansi/index.js 161 bytes {main} [built]
[./node_modules/webpack/hot sync ^\.\/log$] (webpack)/hot sync nonrecursive ^\.\/log$ 170 bytes {main} [built]
[./src/javascripts/helpers/router.js] 1.03 KiB {main} [built]
[./src/javascripts/main.js] 199 bytes {main} [built]
[./src/styles/main.scss] 2.8 KiB {main} [built] [failed] [1 error]
    + 45 hidden modules

ERROR in ./src/styles/main.scss
Module build failed (from ./node_modules/mini-css-extract-plugin/dist/loader.js):
ModuleBuildError: Module build failed (from ./node_modules/sass-loader/dist/cjs.js):
Error: Node Sass does not yet support your current environment: OS X 64-bit with Unsupported runtime (88)
For more information on which environments are supported please see:
https://github.com/sass/node-sass/releases/tag/v4.14.1
    at module.exports (/Users/mjsisk/workspace3/foundations/inclass/IN-CLASS-almost-amazon-template/node_modules/node-sass/lib/binding.js:13:13)
    at Object.<anonymous> (/Users/mjsisk/workspace3/foundations/inclass/IN-CLASS-almost-amazon-template/node_modules/node-sass/lib/index.js:14:35)
    at Module._compile (node:internal/modules/cjs/loader:1091:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1120:10)
    at Module.load (node:internal/modules/cjs/loader:971:32)
    at Function.Module._load (node:internal/modules/cjs/loader:812:14)
    at Module.require (node:internal/modules/cjs/loader:995:19)
    at require (node:internal/modules/cjs/helpers:92:18)
    at getDefaultSassImplementation (/Users/mjsisk/workspace3/foundations/inclass/IN-CLASS-almost-amazon-template/node_modules/sass-loader/dist/utils.js:43:10)
    at getSassImplementation (/Users/mjsisk/workspace3/foundations/inclass/IN-CLASS-almost-amazon-template/node_modules/sass-loader/dist/utils.js:57:30)
    at /Users/mjsisk/workspace3/foundations/inclass/IN-CLASS-almost-amazon-template/node_modules/webpack/lib/NormalModule.js:316:20
    at /Users/mjsisk/workspace3/foundations/inclass/IN-CLASS-almost-amazon-template/node_modules/loader-runner/lib/LoaderRunner.js:367:11
    at /Users/mjsisk/workspace3/foundations/inclass/IN-CLASS-almost-amazon-template/node_modules/loader-runner/lib/LoaderRunner.js:233:18
    at runSyncOrAsync (/Users/mjsisk/workspace3/foundations/inclass/IN-CLASS-almost-amazon-template/node_modules/loader-runner/lib/LoaderRunner.js:143:3)
    at iterateNormalLoaders (/Users/mjsisk/workspace3/foundations/inclass/IN-CLASS-almost-amazon-template/node_modules/loader-runner/lib/LoaderRunner.js:232:2)
    at Array.<anonymous> (/Users/mjsisk/workspace3/foundations/inclass/IN-CLASS-almost-amazon-template/node_modules/loader-runner/lib/LoaderRunner.js:205:4)
    at Storage.finished (/Users/mjsisk/workspace3/foundations/inclass/IN-CLASS-almost-amazon-template/node_modules/enhanced-resolve/lib/CachedInputFileSystem.js:55:16)
    at /Users/mjsisk/workspace3/foundations/inclass/IN-CLASS-almost-amazon-template/node_modules/enhanced-resolve/lib/CachedInputFileSystem.js:91:9
    at /Users/mjsisk/workspace3/foundations/inclass/IN-CLASS-almost-amazon-template/node_modules/graceful-fs/graceful-fs.js:123:16
    at FSReqCallback.readFileAfterClose [as oncomplete] (node:internal/fs/read_file_context:75:3)
 @ ./src/javascripts/main.js 4:0-29
Child HtmlWebpackCompiler:
     1 asset
    Entrypoint HtmlWebpackPlugin_0 = __child-HtmlWebpackPlugin_0
    [./node_modules/html-webpack-plugin/lib/loader.js!./src/index.html] 320 bytes {HtmlWebpackPlugin_0} [built]
Child mini-css-extract-plugin node_modules/css-loader/dist/cjs.js??ref--7-1!node_modules/sass-loader/dist/cjs.js??ref--7-2!src/styles/main.scss:
    Entrypoint mini-css-extract-plugin = *
    [./node_modules/css-loader/dist/cjs.js?!./node_modules/sass-loader/dist/cjs.js?!./src/styles/main.scss] ./node_modules/css-loader/dist/cjs.js??ref--7-1!./node_modules/sass-loader/dist/cjs.js??ref--7-2!./src/styles/main.scss 1.3 KiB {mini-css-extract-plugin} [built] [failed] [1 error]
    
    ERROR in ./src/styles/main.scss (./node_modules/css-loader/dist/cjs.js??ref--7-1!./node_modules/sass-loader/dist/cjs.js??ref--7-2!./src/styles/main.scss)
    Module build failed (from ./node_modules/sass-loader/dist/cjs.js):
    Error: Node Sass does not yet support your current environment: OS X 64-bit with Unsupported runtime (88)
    For more information on which environments are supported please see:
    https://github.com/sass/node-sass/releases/tag/v4.14.1
        at module.exports (/Users/mjsisk/workspace3/foundations/inclass/IN-CLASS-almost-amazon-template/node_modules/node-sass/lib/binding.js:13:13)
        at Object.<anonymous> (/Users/mjsisk/workspace3/foundations/inclass/IN-CLASS-almost-amazon-template/node_modules/node-sass/lib/index.js:14:35)
        at Module._compile (node:internal/modules/cjs/loader:1091:14)
        at Object.Module._extensions..js (node:internal/modules/cjs/loader:1120:10)
        at Module.load (node:internal/modules/cjs/loader:971:32)
        at Function.Module._load (node:internal/modules/cjs/loader:812:14)
        at Module.require (node:internal/modules/cjs/loader:995:19)
        at require (node:internal/modules/cjs/helpers:92:18)
        at getDefaultSassImplementation (/Users/mjsisk/workspace3/foundations/inclass/IN-CLASS-almost-amazon-template/node_modules/sass-loader/dist/utils.js:43:10)
        at getSassImplementation (/Users/mjsisk/workspace3/foundations/inclass/IN-CLASS-almost-amazon-template/node_modules/sass-loader/dist/utils.js:57:30)
ℹ ｢wdm｣: Failed to compile.

**Reply by drteresavasquez** [reply]
Ok. For this check your node version. You should be on 14.x if it's higher than that you installed the wrong version of node. Uninstall and do the LTS version as stated in the machine setup. 

PLEASE do not updated or mess with your environment once you get this fixed. You're doing more harm than good by doing that and now you may not be ready for tonight.

**Comment by drteresavasquez** [comment]
Here is a ticket to help with the sass issue. https://github.com/nss-evening-cohort-14/evening-client-side/discussions/98

**Reply by fimoefive** [reply]
➜  ~ node -v
v15.10.0
➜  ~ npm --version
7.5.3

I ran these command lines on the stack overflow site and everything is compiling again!!! Thank GOD!
npm uninstall node-sass
npm i sass

But I'm still getting the same cookie console issues as I was before.
Indicate whether to send a cookie in a cross-site request by specifying its SameSite attribute

**Reply by fimoefive** [reply]
I reinstalled node-v14. My current versions are.
➜  ~ node -v
v14.16.0
➜  ~ npm --version
6.14.11

I now have down graded the node and npm version. I'm gonna close down all the apps and restart my Mac. Hopefully this will resolve the issues console tab.

**Reply by drteresavasquez** [reply]
How about checking chrome. It's an issue with your browser. Also if the site compiles and you're getting warnings just ignore them.

**Reply by fimoefive** [reply]
**It is completely compiling without any errors and I am rendering the Google Login button in the DOM.**
        + 4 hidden modules
ℹ ｢wdm｣: Compiled successfully.

**Is lockfileVersion@1 a problem? I've also tried serving the port in Safari and FireFox and the server times out also after clicking the Google Login button.**

➜  IN-CLASS-almost-amazon-template git:(main) ✗ npm i
npm WARN read-shrinkwrap This version of npm is compatible with lockfileVersion@1, but package-lock.json was generated for lockfileVersion@2. I'll try to do my best with it!
npm WARN webpack-intro@1.0.0 No repository field.

audited 1195 packages in 4.659s

67 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities

**This is the console error I'm getting:**

auth.js?2ce1:67 Uncaught t {code: "auth/network-request-failed", message: "A network error (such as timeout, interrupted connection or unreachable host) has occurred.", a: null}

**Reply by drteresavasquez** [reply]
That's your internet as @ajawashington told you on the last ticket I think.

**Reply by fimoefive** [reply]
**I'm currently at a coffee shop and getting which is way faster than what I have at home.**

221.3

Mbps download

22.0

Mbps upload

Latency: 45 ms
Server: Atlanta
Your Internet connection is very fast.

Your Internet connection should be able to handle multiple devices streaming HD videos, video conferencing, and gaming at the same time.

**How do I update the attribute cookie?** 

Because a cookie’s SameSite attribute was not set or is invalid, it defaults to SameSite=Lax, which prevents the cookie from being sent in a cross-site request. This behavior protects user data from accidentally leaking to third parties and cross-site request forgery.

Resolve this issue by updating the attributes of the cookie:
Specify SameSite=None and Secure if the cookie should be sent in cross-site requests. This enables third-party use.
Specify SameSite=Strict or SameSite=Lax if the cookie should not be sent in cross-site requests.

**Reply by fimoefive** [reply]
I've reset my Chrome settings to default settings and now the issues tab is no longer displaying in the console, but I am still getting this auth.js console error. 


  | eval | @ | auth.js?2ce1:256
-- | -- | -- | --
  | setTimeout (async) |   |  
  | mc | @ | auth.js?2ce1:67
  | eval | @ | auth.js?2ce1:76
  | tc | @ | auth.js?2ce1:69
  | Promise.then (async) |   |  
  | qc | @ | auth.js?2ce1:69
  | pc | @ | auth.js?2ce1:69
  | Lc | @ | auth.js?2ce1:75
  | vc | @ | auth.js?2ce1:74
  | eval | @ | auth.js?2ce1:69
  | eval | @ | auth.js?2ce1:256
  | setTimeout (async) |   |  
  | eval | @ | auth.js?2ce1:256
  | (anonymous) | @ | cb=gapi.loaded_0:246
  | e.Hq | @ | cb=gapi.loaded_0:217
  | Yk | @ | cb=gapi.loaded_0:220
  | Tk | @ | cb=gapi.loaded_0:220
  | _.Ek.YQ | @ | cb=gapi.loaded_0:220
  | Dk | @ | cb=gapi.loaded_0:212
  | Promise.then (async) |   |  
  | yk | @ | cb=gapi.loaded_0:212
  | _.Ck | @ | cb=gapi.loaded_0:212
  | Uk | @ | cb=gapi.loaded_0:219
  | _.Ek.Vg | @ | cb=gapi.loaded_0:218
  | (anonymous) | @ | cb=gapi.loaded_0:213
  | _.pm.open | @ | cb=gapi.loaded_0:260
  | eval | @ | auth.js?2ce1:256
  | C | @ | auth.js?2ce1:69
  | eval | @ | auth.js?2ce1:256
  | e.g | @ | auth.js?2ce1:73
  | Oc | @ | auth.js?2ce1:76
  | Kc | @ | auth.js?2ce1:76
  | k.fc | @ | auth.js?2ce1:75
  | tc | @ | auth.js?2ce1:69
  | Promise.then (async) |   |  
  | qc | @ | auth.js?2ce1:69
  | pc | @ | auth.js?2ce1:69
  | Lc | @ | auth.js?2ce1:75
  | vc | @ | auth.js?2ce1:74
  | eval | @ | auth.js?2ce1:69
  | (anonymous) | @ | api.js?onload=__iframefcb543294:15
  | fa | @ | api.js?onload=__iframefcb543294:8
  | A | @ | api.js?onload=__iframefcb543294:15
  | r.<computed> | @ | api.js?onload=__iframefcb543294:15
  | x.<computed> | @ | api.js?onload=__iframefcb543294:15
  | (anonymous) | @ | cb=gapi.loaded_0:1

**Reply by fimoefive** [reply]
HOLLY MOLLY!!!! I just got it to sign in and render in Safari, but still no dice in Google CHROME!!!!!!!!!!!!!!!

I had to close VSC and refire up npm start for it to work.

**Comment by fimoefive** [comment, marked as answer]
Thank You DR. T and AJA!!! I'm still not sure what the issue was though, but thank ya'll both for helping me.
Also sorry for being a pain and thanks for for helping me out!!

**Reply by drteresavasquez** [reply]
As I said in class, you're NOT a pain! You're finding issues that are a part of the dev process so it's all a learning process. Glad you got something working. On Saturday let's take a closer look at your chrome issue to see if we can get those resolved. We just had a full night tonight but Saturday should give us more time.
